src/components/PostDetails.tsx

```diff
@@ -1,117 +1,120 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Post } from '../types/Post';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Comment } from '../types/Comment';
+import { getComments, removeComment } from '../api/comments';
+import { CommentCard } from './CommentCard/CommentCard';
+
+type Props = {
+  post: Post;
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[] | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isNewCommentForm, setIsNewCommentForm] = useState(false);
+
+  const loadComments = (postId: number) => {
+    getComments(postId)
+      .then(data => {
+        setComments(data);
+      })
+      .catch(() => {
+        setIsError(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  };
+
+  useEffect(() => {
+    setIsLoading(true);
+    setIsError(false);
+
+    loadComments(post.id);
+  }, [post.id]);
+
+  const deleteComment = (commentId: number) => {
+    setIsLoading(true);
+    setIsError(false);
+
+    removeComment(commentId)
+      .then(() => {
+        if (comments) {
+          const newComments: Comment[] = comments
+            .filter(c => c.id !== commentId);
+
+          setComments(newComments);
+        }
+      })
+      .catch(() => {
+        setIsError(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  };
```

Comment:
The comment should be deleted using the optimistic scenario (you delete a comment from the comments state, then send a request to the server, and only if the request fails, you return the deleted comment to the comments state)